include/response_reader.rb

```diff
@@ -105,10 +105,10 @@ def url_and_args_from_location
     url << "?#{uri.query}" if uri.query.to_s.length > 0
     url = url.gsub(/\A\//, "")
 
-    args = {host: uri.host}
+    args = @http2.args
+           .except(:port)
```

Comment:
I guess SSL should also be included?

I am thinking you start out with SSL in args, then get redirected to a site without, but SSL will be set to true, because the parent connection had it?